### PR TITLE
Add NeuralNetwork constructor to load from a trained .pt file

### PR DESCRIPTION
**Pull request by AiRuiChen**
Tested: this code is tested to work w/ the MPC neural network (generated [here](https://github.com/uscresl/hybrid-sim-exp/tree/master/neural_mpc)) we used for the Laikago locomotion experiment.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,22 @@ else(Eigen3_FOUND)
     endif(USE_EIGEN_FALLBACK)
 endif (Eigen3_FOUND)
 
+OPTION(USE_TORCH "Use LibTorch" OFF)
+IF (USE_TORCH)
+    find_package(Torch)
+    if (Torch_FOUND)
+        message("FOUND TORCH")
+        message("Torch include directories: ${TORCH_LIBRARIES}")
```
Comment (eric-heiden) [inline CMakeLists.txt]: The variable corresponds to libraries, not include directories

File: src/math/neural_network.hpp [hunk 2]
```diff
@@ -261,7 +265,7 @@ class NeuralNetworkSpecification {
             current[ci] = Algebra::log(one + Algebra::exp(current[ci]));
             break;
           case NN_ACT_ELU:
-            current[ci] = tds::where_ge(current[ci], zero, current[ci],
+            current[ci] = tds::where_gt(current[ci], zero, current[ci],
```
Comment (eric-heiden) [inline src/math/neural_network.hpp]: Is this change necessary?

File: src/math/neural_network.hpp [hunk 3]
```diff
@@ -378,6 +382,63 @@ class NeuralNetwork : public NeuralNetworkSpecification {
   explicit NeuralNetwork(const NeuralNetworkSpecification& spec)
       : NeuralNetworkSpecification(spec) {}
 
+#if USE_TORCH
```
Comment (eric-heiden) [inline src/math/neural_network.hpp]: use `#ifdef USE_TORCH`

File: src/math/neural_network.hpp [hunk 4]
```diff
@@ -410,6 +471,46 @@ class NeuralNetwork : public NeuralNetworkSpecification {
         ->template save_graphviz<Algebra>(filename, input_names, output_names,
                                           weights, biases);
   }
+ private:
+#if USE_TORCH
```
Comment (eric-heiden) [inline src/math/neural_network.hpp]: use `#ifdef USE_TORCH`

**Review by eric-heiden — CHANGES_REQUESTED** [review]
Thanks for the PR! Could you address my comments, and also use `.cpp` as extension for the demo you implemented (like we have the other examples).

File: src/math/neural_network.hpp (hunk 2 above)
Comment (AiRuiChen) [reply src/math/neural_network.hpp]: It's effectively the same. Reverted.

**Comment by AiRuiChen** [comment]
Closing this as I will directly create a PR to the google-research repo

**Pull request closed without merging**